src/Table.php

```diff
@@ -55,12 +56,12 @@ public function addPrimaryKey(string|array $columns)
      * auto increment attribute
      *
      * @param string|\Phinx\Db\Table\Column $columnName Column Name
-     * @param string|\Phinx\Util\Literal|null $type Column Type
+     * @param string|\Phinx\Util\Literal $type Column Type
      * @param array $options Column Options
      * @throws \InvalidArgumentException
      * @return $this
      */
-    public function addColumn($columnName, $type = null, $options = [])
+    public function addColumn(Column|string $columnName, $type, $options = [])
```

Comment:
I think the type should stay nullable, just like in https://github.com/cakephp/phinx/pull/2218#discussion_r1333812072. When a `Column` object is provided, that object can hold the type.

Reply:
Ok. The null was dropped because some tests were passing incompatible values. I don't see this in function in my PR comments, but a couple nullable types were dropped because they seemed to only allow invalid text data (string name + null type).